### PR TITLE
Fix --dry-run interactions

### LINKED ISSUE #2748: renew: letsencrypt.client:Registering without email!
I succesfully set up a certificate for https://turbolab.it/ (CentOS 6.7 with Apache 2.2) like this:

````
git clone https://github.com/letsencrypt/letsencrypt /opt/letsencrypt
cd /opt/letsencrypt
./letsencrypt-auto certonly --webroot -w /full/site/path/ -d turbolab.it -d www.turbolab.it
````

I provided my email + agreed to TOS from the *graphic* (blue) UI.

**Now I'm trying to stage the cert renewal but I'm facing issue**. When I try to renew with `./letsencrypt-auto renew --dry-run` I get:

```
[root@skycracker letsencrypt]# ./letsencrypt-auto renew --dry-run
Checking for new version...
Creating virtual environment...
Installing Python packages...
Installation succeeded.
Requesting root privileges to run letsencrypt...
   /root/.local/share/letsencrypt/bin/letsencrypt renew --dry-run
/root/.local/share/letsencrypt/lib/python2.6/site-packages/cryptography/__init__.py:26: DeprecationWarning: Python 2.6 is no longer supported by the Python core team, please upgrade your Python. A future version of cryptography will drop support for Python 2.6
  DeprecationWarning
Processing /etc/letsencrypt/renewal/turbolab.it.conf
2016-04-02 13:49:45,115:WARNING:letsencrypt.client:Registering without email!
2016-04-02 13:49:45,910:WARNING:letsencrypt.cli:Attempting to renew cert from /etc/letsencrypt/renewal/turbolab.it.conf produced an unexpected error: Missing command line flag or config entry for this setting:
Please read the Terms of Service at https://letsencrypt.org/documents/LE-SA-v1.0.1-July-27-2015.pdf. You must agree in order to register with the ACME server at https://acme-staging.api.letsencrypt.org/directory

(You can set this with the --agree-tos flag). Skipping.
** DRY RUN: simulating 'letsencrypt renew' close to cert expiry
**          (The test certificates below have not been saved.)

All renewal attempts failed. The following certs could not be renewed:
  /etc/letsencrypt/live/turbolab.it/fullchain.pem (failure)
** DRY RUN: simulating 'letsencrypt renew' close to cert expiry
**          (The test certificates above have not been saved.)
1 renew failure(s), 0 parse failure(s)
```

The `DeprecationWarning` makes me very uncomfortable, but I understood shouldn't be an issue.

The failure is the real issue here. I understood that the `renew` parameter should fire a renew with the same options used at install-time, so why the failure?

Thanks!

Update: I can confirm that `/etc/letsencrypt/renewal/turbolab.it.conf` holds the correct setting `
email = info@ MY SITE`

### PR DESCRIPTION
**Pull request by bmw**
Fixes #2748.

### CODE REVIEW

**Comment by bmw** [comment]
@jsha: holy lookup errors batman - [log1](https://gist.github.com/102d8883cd784d38dcaa967694d108d2), [log2](https://gist.github.com/991f50875145ad073d44c2e58550c8d8), [log3](https://gist.github.com/a18d3fcf46cebc2e40196857280d7d55)

**Comment by jsha** [comment]
Sorry, this is my bad - I merged a Boulder change that broke your build. PR to unbreak build in 3, 2, 1...

File: letsencrypt/client.py [hunk 1]
```diff
@@ -105,7 +105,8 @@ def register(config, account_storage, tos_cb=None):
                    "--register-unsafely-without-email was not present.")
             logger.warn(msg)
             raise errors.Error(msg)
-        logger.warn("Registering without email!")
+        if not config.dry_run:
```
Comment (pde) [inline letsencrypt/client.py]: Channeling Seth a little here, there are subtleties to this:
* Should we also suppress this message in staging mode?  Or do we expect people to know about their staging accounts?
* If not, should we note when what we're registering for is a staging account?

**Comment by pde** [comment]
Overall, LGTM, merge when green.

**Comment by bmw** [comment]
Ah great! Thanks for the quick PR jsha.

File: letsencrypt/client.py (hunk 1 above)
Comment (bmw) [inline letsencrypt/client.py]: > Should we also suppress this message in staging mode? Or do we expect people to know about their staging accounts?

I don't have a strong opinion on this either way. If we want to suppress this message when using the staging server, we can simply change this line to `if not config.dry_run` (and fix the test that will break).

> If not, should we note when what we're registering for is a staging account?

I don't think so. Treating servers differently in places other than `cli.py` seems clunky and unnecessary. The only way we're going to ask a user to register for a staging account is if they specifically requested it with the CLI/config file. Given this, adding a special prompt here doesn't seem like a big gain.

**Pull request merged**